FluentTc/Engine/ProjectsRetriever.cs

```diff
@@ -30,19 +31,19 @@ internal class ProjectsRetriever : IProjectsRetriever
         public IList<Project> GetProjects(Action<IBuildProjectHavingBuilder> having = null)
         {
             var locator = having == null ? string.Empty : GetLocator(having);
-            return m_TeamCityCaller.GetFormat<ProjectWrapper>("/app/rest/projects/{0}", locator).Project;
+            return m_TeamCityCaller.GetFormat<ProjectWrapper>(GetApiCall ("/{0}"), locator).Project;
         }
 
-        private string GetLocator(Action<IBuildProjectHavingBuilder> having)
+        public Project GetProject(string projectId)
         {
-            var buildProjectHavingBuilder = m_BuildProjectHavingBuilderFactory.CreateBuildProjectHavingBuilder();
-            having(buildProjectHavingBuilder);
-            return buildProjectHavingBuilder.GetLocator();
+            return m_TeamCityCaller.GetFormat<Project>(GetApiCall("/id:{0}"), projectId);
         }
 
-        public Project GetProject(string projectId)
+        public Project GetProject(Action<IBuildProjectHavingBuilder> having = null)
```

Comment:
You are right, I was not fully looking at it correctly. I added the logic to make sure we are only returning one and made it similar to GetAgent().